File: Data Structures & Algorithms/two-integer-sum/submission-3.go
func twoSum(nums []int, target int) []int{
	dict := make(map[int]int)
	for i := range nums{
		_, ok := dict[target - nums[i]]
		if ok {
			return []int{dict[target-nums[i]], i}
		}
 		dict[nums[i]] = i
	}

	return []int{}
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if len(numbers) == 2:
            return [1, 2]

        i, j = 0, len(numbers)-1

        while i < j:
            if target > (numbers[i] + numbers[j]):
                i += 1
            
            if target < (numbers[i] + numbers[j]):
                j -= 1
            
            if target == (numbers[i] + numbers[j]):
                return [i+1, j+1]
        
        return []